notification-service/pom.xml

```diff
@@ -157,6 +210,24 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.springdoc</groupId>
+                <artifactId>springdoc-openapi-maven-plugin</artifactId>
+                <version>1.4</version>
+                <configuration>
+                    <outputFileName>api-docs.json</outputFileName>
+                    <outputDir>${basedir}/generated/openapi</outputDir>
+                </configuration>
+                <executions>
+                <execution>
+                <id>generate-openapi-json</id>
+                    <goals>
+                        <goal>generate</goal>
+                    </goals>
+                    <phase>package</phase>
```

Comment:
I moved the `generate-client-api-code` plugin to `notification-service` (instead of `notification-service-sdk`), in phase `pre-integration-test`. `generate-openapi-json` plugin must be executed in earlier phase, therefore its phase is `package` and can't be changed to `verify`.
Anyway, I think that the verification in the CI that files are not changed is good idea.